ledger/complete/wal/checkpoint_v6_reader.go

```diff
@@ -0,0 +1,340 @@
+package wal
+
+import (
+	"bufio"
+	"fmt"
+	"io"
+	"os"
+	"path"
+
+	"github.com/onflow/flow-go/ledger/complete/mtrie/flattener"
+	"github.com/onflow/flow-go/ledger/complete/mtrie/node"
+	"github.com/onflow/flow-go/ledger/complete/mtrie/trie"
+)
+
+// ReadCheckpointV6 reads checkpoint file from a main file and 17 file parts.
+// the main file stores:
+// 		1. version
+//		2. checksum of each part file (17 in total)
+// 		3. checksum of the main file itself
+// 	the first 16 files parts contain the trie nodes below the subtrieLevel
+//	the last part file contains the top level trie nodes above the subtrieLevel and all the trie root nodes.
+func ReadCheckpointV6(dir string, fileName string) ([]*trie.MTrie, error) {
```

Comment:
Since it is concurrent now, you can consider adding tracing to the whole state load/store codepath so you can visualize in grafana (for example right now it would likely show that trie construction is the bottleneck).

Reply:
Yeah, I will be adding logs in this function in future PR. I think logs would be more useful than tracing. 

Since we will only read the checkpoint once on every restart, it's OK to print info logs with each key steps during the read.